src/mtev_http.c

```diff
@@ -1449,7 +1450,6 @@ mtev_http_session_drive(eventer_t e, int origmask, void *closure,
     } else {
 #endif
       _http_perform_write(ctx, &maybe_write_mask);
-      if(ctx->conn.e == NULL) goto release;
```

Comment:
are you sure this should be removed.  I thought _http_perform_write could potentially destroy the connection and that was why it was immediately checked after this call.